app/models/task.rb

```diff
@@ -0,0 +1,71 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  activity_message_value    :string(65536)    not null
+#  id                        :bigint(8)        not null, primary key
+#  job_class                 :string(255)      not null
+#  job_enqueued_for_retry_at :datetime
+#  job_first_started_at      :datetime
+#  job_last_failed_at        :datetime
+#  job_succeeded_at          :datetime
+#  job_type_value            :string(255)      not null
+#  num_attempts              :integer          default(0), not null
+#  provider_job_id           :string
+#
+
+class Task < ApplicationRecord
+  def enqueue(job_params: {})
+    job = job_class.constantize.perform_later(job_params.merge(task_id: id))
```

Comment:
Not totally sure about this approach for job params.

Reply:
Me either, but I don't have a better one, let's just try working with it and see where it leaves us. If it ends up being a pain in the neck to work with in practice, we can revisit. Sometimes you just need to try things.